### PR TITLE
C++: Fix `chmod` prototype in toctou test and additional test

### PR DESCRIPTION
**Pull request by jketema**

### CODE REVIEW

File: cpp/ql/test/query-tests/Security/CWE/CWE-367/semmle/test2.cpp [hunk 1]
```diff
@@ -408,3 +408,8 @@ void test7_1(const char *path1, const char *path2)
 		chmod(path2, 1234); // BAD
 	}
 }
+
+int test8(const char *path, int mode)
+{
+    return (chmod(path, mode) == 0 ? 1 : 0);
```
Comment (jketema) [inline cpp/ql/test/query-tests/Security/CWE/CWE-367/semmle/test2.cpp:414]: I'd actually like to know if this if `GOOD` or `BAD` 😅

Comment (MathiasVP) [reply cpp/ql/test/query-tests/Security/CWE/CWE-367/semmle/test2.cpp:414]: This looks good to me! There's a sensitive file operation here, `chmod`, which represents the "time of use" part of the query, but there's no check that guards the operation (i.e., there's no "time of check").

Comment (jketema) [reply cpp/ql/test/query-tests/Security/CWE/CWE-367/semmle/test2.cpp]: Annotation added.

**Review by MathiasVP — APPROVED** [review]
LGTM!

**Pull request merged**